addons/purchase/models/purchase.py

```diff
@@ -530,6 +530,15 @@ def action_create_invoice(self):
         for vals in invoice_vals_list:
             moves |= AccountMove.with_company(vals['company_id']).create(vals)
 
+        analytic_account = self.env['account.analytic.account']
+        for move_index in range(len(moves)):
+            for line_index in range(len(moves[move_index].invoice_line_ids)):
+                invoice_line_ids = invoice_vals_list[move_index]['invoice_line_ids']
+                analytic_account_id = invoice_line_ids[line_index][2]['analytic_account_id']
+                po_account = analytic_account.browse(analytic_account_id)
```

Comment:
What is the reason this can not be done in the _prepare_account_move_line?

Reply:
It's seems to be already there. https://github.com/odoo/odoo/blob/14.0/addons/purchase/models/purchase.py#L1142
I agree with jco. You should check or at least explain why it's not working that way